### PR TITLE
 add integration tests for linux and windows, war and jar deploy

### LINKED ISSUE #211: Better naming for integration tests
Make the folder and a prefix to pointing out its number order: for example: `1.windows-ftp`

### PR DESCRIPTION
**Pull request by SummerSun**
fix #211 #212

### CODE REVIEW

File: azure-webapp-maven-plugin/src/it/linux-jar/cleanup.groovy [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ * license information.
+ */
+
+// Clean up resources created in test
+def clientId = System.getenv("CLIENT_ID")
+def tenantId = System.getenv("TENANT_ID")
+def key = System.getenv("KEY")
+def command = """
+    az login --service-principal -u ${clientId}  -p ${key} --tenant ${tenantId}
+    az group delete -y -n maven-webapp-it-rg-1 --no-wait
```
Comment (jdneo) [inline azure-webapp-maven-plugin/src/it/linux-jar/cleanup.groovy]: `maven-webapp-it-rg-1`: the suffix `1` might be used by other cases. Let's use a unique suffix for each test cases.

File: azure-webapp-maven-plugin/src/it/linux-jar/pom.xml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        <plugins>
+            <plugin>
+                <groupId>@project.groupId@</groupId>
+                <artifactId>@project.artifactId@</artifactId>
+                <version>@project.version@</version>
+                <configuration>
+                    <authentication>
+                        <serverId>azure-auth</serverId>
+                    </authentication>
+                    <resourceGroup>maven-webapp-it-rg-1</resourceGroup>
```
Comment (jdneo) [inline azure-webapp-maven-plugin/src/it/linux-jar/pom.xml]: unique suffix

File: azure-webapp-maven-plugin/src/it/linux-jar/setup.groovy [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+ * license information.
+ */
+
+// Delete resources before test
+def clientId = System.getenv("CLIENT_ID")
+def tenantId = System.getenv("TENANT_ID")
+def key = System.getenv("KEY")
+def command = """
+    az login --service-principal -u ${clientId}  -p ${key} --tenant ${tenantId}
+    az group delete -y -n maven-webapp-it-rg-1
```
Comment (jdneo) [inline azure-webapp-maven-plugin/src/it/linux-jar/setup.groovy]: unique suffix

File: azure-webapp-maven-plugin/src/it/linux-war/cleanup.groovy [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+ * license information.
+ */
+
+// Clean up resources created in test
+def clientId = System.getenv("CLIENT_ID")
+def tenantId = System.getenv("TENANT_ID")
+def key = System.getenv("KEY")
+def command = """
+    az login --service-principal -u ${clientId}  -p ${key} --tenant ${tenantId}
+    az group delete -y -n maven-webapp-it-rg-5 --no-wait
```
Comment (jdneo) [inline azure-webapp-maven-plugin/src/it/linux-war/cleanup.groovy]: unique suffix

File: azure-webapp-maven-plugin/src/it/linux-war/pom.xml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        <plugins>
+            <plugin>
+                <groupId>@project.groupId@</groupId>
+                <artifactId>@project.artifactId@</artifactId>
+                <version>@project.version@</version>
+                <configuration>
+                    <authentication>
+                        <serverId>azure-auth</serverId>
+                    </authentication>
+                    <resourceGroup>maven-webapp-it-rg-5</resourceGroup>
```
Comment (jdneo) [inline azure-webapp-maven-plugin/src/it/linux-war/pom.xml]: unique suffix

File: azure-webapp-maven-plugin/src/it/linux-war/setup.groovy [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+ * license information.
+ */
+
+// Delete resources before test
+def clientId = System.getenv("CLIENT_ID")
+def tenantId = System.getenv("TENANT_ID")
+def key = System.getenv("KEY")
+def command = """
+    az login --service-principal -u ${clientId}  -p ${key} --tenant ${tenantId}
+    az group delete -y -n maven-webapp-it-rg-5
```
Comment (jdneo) [inline azure-webapp-maven-plugin/src/it/linux-war/setup.groovy]: unique suffix

**Comment by SummerSun** [comment]
OS: Windows 10
Mark them here so we could have further discussion and update the solution.

Run the IT in powershell/git bash/console

Currently, only this way the setup and cleanup build scripts could run correctly and azure resources get cleaned up after testing.

```
def command1 = "az login --service-principal -u ${clientId}  -p ${key} --tenant ${tenantId}"
def command2 = "az group delete -y -n maven-webapp-it-rg-8"
def command3 = "az logout"

if (System.properties['os.name'].contains('Windows')) {
    println "cmd /c ${command1}".execute().text
    println "cmd /c ${command2}".execute().text
    println "cmd /c ${command3}".execute().text
} else {
    println "bash -c ${command1}".execute().text
    println "bash -c ${command2}".execute().text
    println "bash -c ${command3}".execute().text
}
```

Following two solutions does not work and we bear have any error log.

* Triple quotes DTW
test run correctly, but azure resources are neither cleaned up in setup nor in cleanup script

```
def command = """
    az login --service-principal -u ${clientId}  -p ${key} --tenant ${tenantId}
    az group delete -y -n maven-webapp-it-rg-8 --no-wait
    az logout
"""
def process
if (System.properties['os.name'].contains('Windows')) {
    process = ["cmd", "/c", command].execute()
} else {
    process = ["bash", "-c", command].execute()
}

println process.text
```
* array.each DTW
test does not run correctly, post build script run immediately after pre build script, no deploy task performed
```
def commands = [command1, command2, command3] as String[]

if (System.properties['os.name'].contains('Windows')) {
    commands.each {
        println "cmd /c ${it}".execute().text
    }
} else {
    commands.each {
        println "bash -c ${it}".execute().text
    }
}
```

**Pull request merged**